src/Delayed/operators.jl

```diff
@@ -0,0 +1,56 @@
+
+abstract type Operator <: Types.AbstractDelayed end
+
+"""
+$(TYPEDEF)
+$(TYPEDFIELDS)
+
+"""
+struct BinaryOperator <: Types.AbstractDelayed
+    left::NestedReal
+    right::NestedReal
+    operator::Function
+end
+
+
+"""
+$(TYPEDEF)
+$(TYPEDFIELDS)
+
+"""
+struct UnaryOperator <: Types.AbstractDelayed
+    operand::NestedReal
+    operator::Function
+end
+
+
+
+Base.:+(left::Types.AbstractDelayed, right::Types.AbstractDelayed) = BinaryOperator(left, right, +)
+Base.:+(left::Types.AbstractDelayed, right::Real) = BinaryOperator(left, right, +)
+Base.:+(left::Real, right::Types.AbstractDelayed) = BinaryOperator(left, right, +)
+
+
+Base.:-(left::Types.AbstractDelayed, right::Types.AbstractDelayed) = BinaryOperator(left, right, -)
+Base.:-(left::Types.AbstractDelayed, right::Real) = BinaryOperator(left, right, -)
+Base.:-(left::Real, right::Types.AbstractDelayed) = BinaryOperator(left, right, -)
+Base.:-(operand::Types.AbstractDelayed) = UnaryOperator(operand, -)
```

Comment:
why do we have this UnaryOperator(operand, -) only for subtraction ?

Reply:
The unary operator was meant to enable greater flexibility of expressions by allowing abs, ceil, round, etc these kind of functions. I didn't feel comfortable producing a list of such functions to overload and minus is the only arithmetic operator off the top of my head that this applies to. If you think something is missing do let me know

Reply:
I found out round will be used for this space, but still I could do it with other changes you made 
space = Dict(
                 :min_data_in_leaf =>TreeParzen.Delayed.UnaryOperator(1007160 * (10^HP.QuantUniform(:min_data_in_leaf, -5., -2., (3/19))), round))
but maybe this is not the efficient way as it will cast all the sample even if they are not going to be selected